src/Microsoft.DotNet.Tools.Builder/CompileContext.cs

```diff
@@ -353,14 +353,15 @@ public static CompilerIO GetCompileIO(ProjectContext project, string config, str
             // input: key file
             AddKeyFile(project, config, compilerIO);
 
+            var binariesOutputPath = project.GetOutputPathCalculator(outputPath).GetOutputDirectoryPath(config);
             // output: compiler output
-            compilerIO.Outputs.Add(CompilerUtil.GetCompilationOutput(project.ProjectFile, project.TargetFramework, config, outputPath));
+            compilerIO.Outputs.Add(CompilerUtil.GetCompilationOutput(project.ProjectFile, project.TargetFramework, config, binariesOutputPath));
 
             // input / output: resources without culture
             AddCultureResources(project, intermediaryOutputPath, compilerIO);
 
             // input / output: resources with culture
-            AddNonCultureResources(project, outputPath, compilerIO);
+            AddNonCultureResources(project, binariesOutputPath, compilerIO);
```

Comment:
You'll have to also propagate binariesOutputPath to the changes that went into master today, that also treat the xml and pdb. It involves the same change, replacing outputPath with binariesOutputPath